src/main/java/sh/platform/config/MariaDB.java

```diff
@@ -0,0 +1,37 @@
+package sh.platform.config;
+
+import org.mariadb.jdbc.MariaDbDataSource;
+
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * A credential specialization that provides a {@link DataSource} using MySQL database
+ */
+public class MariaDB extends SQLDatabase implements Supplier<DataSource> {
+
+    static final String DRIVER = "org.mariadb.jdbc.Driver";
+
+    static final String PROVIDER = "mariadb";
+
+    public MariaDB(Map<String, Object> config) {
+        super(config);
+    }
+
+    @Override
+    public DataSource get() {
+        try {
+            MariaDbDataSource dataSource = new MariaDbDataSource();
+            dataSource.setDatabaseName(getPath());
+            dataSource.setUser(getUserName());
+            dataSource.setPassword(getPassword());
+            dataSource.setPort(getPort());
+            dataSource.setServerName(getHost());
+            return dataSource;
+        } catch (SQLException exp) {
```

Comment:
This error could be more explanatory. 

"Invalid credentials: unable to start MariaDB DataSource" ?

Also, I'm noticing that some of the other services don't have a catch block that throw `PlatformshException`, how come?

Reply:
Yes, that is because each client API has your way. I don't like checked exception. But if the MariaDB likes this way